mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/quota/CassandraPerDomainMaxQuotaDao.java

```diff
@@ -0,0 +1,134 @@
+
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.querybuilder.Delete;
+import com.datastax.driver.core.querybuilder.Insert;
+import com.datastax.driver.core.querybuilder.Select;
+
+public class CassandraPerDomainMaxQuotaDao {
+
+    private final Session session;
+    private final PreparedStatement setMaxStorageStatement;
+    private final PreparedStatement setMaxMessageStatement;
+    private final PreparedStatement getMaxStorageStatement;
+    private final PreparedStatement getMaxMessageStatement;
+    private final PreparedStatement removeMaxStorageStatement;
+    private final PreparedStatement removeMaxMessageStatement;
+
+    @Inject
+    public CassandraPerDomainMaxQuotaDao(Session session) {
+        this.session = session;
+        this.setMaxStorageStatement = session.prepare(setMaxStorageStatement());
+        this.setMaxMessageStatement = session.prepare(setMaxMessageStatement());
+        this.getMaxStorageStatement = session.prepare(getMaxStorageStatement());
+        this.getMaxMessageStatement = session.prepare(getMaxMessageStatement());
+        this.removeMaxStorageStatement = session.prepare(removeMaxStorageStatement());
+        this.removeMaxMessageStatement = session.prepare(removeMaxMessageStatement());
+    }
+
+    private Delete.Where removeMaxMessageStatement() {
+        return delete().column(CassandraDomainMaxQuota.MESSAGE_COUNT)
+            .from(CassandraDomainMaxQuota.TABLE_NAME)
+            .where(eq(CassandraDomainMaxQuota.DOMAIN, bindMarker()));
+    }
+
+    private Delete.Where removeMaxStorageStatement() {
+        return delete().column(CassandraDomainMaxQuota.STORAGE)
+            .from(CassandraDomainMaxQuota.TABLE_NAME)
+            .where(eq(CassandraDomainMaxQuota.DOMAIN, bindMarker()));
+    }
+
+    private Select.Where getMaxMessageStatement() {
+        return select(CassandraDomainMaxQuota.MESSAGE_COUNT)
+            .from(CassandraDomainMaxQuota.TABLE_NAME)
+            .where(eq(CassandraDomainMaxQuota.DOMAIN, bindMarker()));
+    }
+
+    private Select.Where getMaxStorageStatement() {
+        return select(CassandraDomainMaxQuota.STORAGE)
+            .from(CassandraDomainMaxQuota.TABLE_NAME)
+            .where(eq(CassandraDomainMaxQuota.DOMAIN, bindMarker()));
+    }
+
+    private Insert setMaxMessageStatement() {
+        return insertInto(CassandraDomainMaxQuota.TABLE_NAME)
+            .value(CassandraDomainMaxQuota.DOMAIN, bindMarker())
+            .value(CassandraDomainMaxQuota.MESSAGE_COUNT, bindMarker());
+    }
+
+    private Insert setMaxStorageStatement() {
+        return insertInto(CassandraDomainMaxQuota.TABLE_NAME)
+            .value(CassandraDomainMaxQuota.DOMAIN, bindMarker())
+            .value(CassandraDomainMaxQuota.STORAGE, bindMarker());
+    }
+
+    public void setMaxStorage(String domain, QuotaSize maxStorageQuota) {
```

Comment:
I think we are not consistent with domains:
in _CassandraDomainList_ we are using domains in lower case (https://github.com/linagora/james-project/blob/a11ce53c06ad97534334562fc583dab7b73d1294/server/data/data-cassandra/src/main/java/org/apache/james/domainlist/cassandra/CassandraDomainList.java#L114)
not in this dao.

Maybe it's time to have a DAO for domains & introduce domainIds

Reply:
Could be done in a separate PR imo.

Until then we can enforce lower case in the different MaxQuotaManagers.

Would that be OK for you?

Reply:
Ok, just create an issue for that point.